### PR TITLE
Outbound Firewall Rules - RestrictOutboundNetworkAccess Powershell

### PR DESCRIPTION
**Pull request by VMMicrosoft**
## Description

Outbound Firewall Rules - Restrict Access - Added parameters to the new and set cmdlets.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - Azure/azure-powershell-cmdlet-review-pr#962
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
hi @VMMicrosoft , CI failed due to some test cases failed, please fix them, thanks.

**Comment by BethanyZhou** [comment]
Hi @VMMicrosoft , please resolve confllicts.

**Comment by BethanyZhou** [comment]
Hi @VMMicrosoft , please raise a design review for the change of cmdlets on https://github.com/Azure/azure-powershell-cmdlet-review-pr.

**Comment by VMMicrosoft** [comment]
Powershell Cmdlet review - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/962

**Comment by VMMicrosoft** [comment]
@BethanyZhou - Ready to go! Please review...

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -33,7 +33,13 @@
     - `Get-AzSqlServerTransparentDataEncryptionProtector`
     - `Set-AzSqlInstanceTransparentDataEncryptionProtector`
     - `Get-AzSqlInstanceTransparentDataEncryptionProtector`
-
+* Added RestrictOutboundNetworkAccess parameter to following cmdlets
```
Comment (BethanyZhou) [inline src/Sql/Sql/ChangeLog.md:36]: Please add change log msg under the top of upcoming release section

Comment (VMMicrosoft) [reply src/Sql/Sql/ChangeLog.md]: Done!

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        /// Gets or sets the allowed FQDN of the Azure Sql Database Server Outbound Firewall Rule
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "The Azure Sql Database Server Outbound Firewall Rule FQDN.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string AllowedFQDN { get; set; }
```
Comment (BethanyZhou) [inline src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs:39]: I suggest to move the definition of AllowedFQDN to AzureSqlServerOutboundFirewallRulesCmdletBase

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                try
+                {
+                    model = ModelAdapter.GetOutboundFirewallRule(this.ResourceGroupName, this.ServerName, this.AllowedFQDN);
+                    results.Add(model);
+                }
+                catch (CloudException ex)
+                {
+                    if (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                    {
+                        throw new PSArgumentException(
```
Comment (BethanyZhou) [inline src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs:63]: We recommend using AzPSArgument*Exception. Please refer to https://github.com/Azure/azure-powershell-common/tree/main/src/Common/Exceptions and find proper exceptions to replace these ones.

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                            string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.ServerOutboundFirewallRuleFQDNDoesNotExist, this.AllowedFQDN, this.ServerName), "AllowedFQDN");
```
Comment (BethanyZhou) [inline src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs:64]:
```suggestion
                            string.Format(Resources.ServerOutboundFirewallRuleFQDNDoesNotExist, this.AllowedFQDN, this.ServerName), "AllowedFQDN");
```
Believe we can use Resources directly.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs (hunk 2 above)
Comment (VMMicrosoft) [reply src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs:39]: The AllowedFQDN parameter is optional in `Get-AzSqlServerOutboundFirewallRules`. Therefore, Mandatory = false for Get but it is true for New and Remove.

Comment (BethanyZhou) [reply src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs:39]: Make sense.

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs (hunk 4 above)
Comment (VMMicrosoft) [reply src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs]: Done. Fixed.

File: src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs (hunk 3 above)
Comment (VMMicrosoft) [reply src/Sql/Sql/OutboundFirewallRules/Cmdlet/GetAzureSqlServerOutboundFirewallRule.cs]: Done. Fixed.

**Comment by VMMicrosoft** [comment]
Ready for reivew.

**Review by BethanyZhou — APPROVED** [review]
LGTM.

**Comment by VMMicrosoft** [comment]
@BethanyZhou - Please merge when ready... :)

**Pull request merged**